plugins/api/discovery/src/org/apache/cloudstack/discovery/ApiDiscoveryServiceImpl.java

```diff
@@ -223,7 +223,9 @@ private ApiDiscoveryResponse getCmdRequestMap(Class<?> cmdClass, APICommand apiC
 
             for (APIChecker apiChecker : _apiAccessCheckers) {
                 try {
-                    apiChecker.checkAccess(user, name);
+                    if (!"ApiRateLimitServiceImpl".equals(apiChecker.getName())) {
```

Comment:
you can use something like this.class.getName() or the .getName()... instead of the hard coded string comparison.

Reply:
Using the 'ApiRateLimitServiceImpl' class instead of a hard coded string introduces the rate limit plugin module as a dependency to the api discovery module. I didn't think that was a good idea.

Reply:
What if you used the command class instead? I mean, you are using the "service class", but at this point here you have the command that was used to list the APIs, right?  You can use it, and then you do not need to use this hardcoded String.

Reply:
Ok, I am now seeing the big picture. I looking at the code here. I will provide some feedback in another comment